server/src/resolverMap.ts

```
@@ -0,0 +1,120 @@
+];
+
+const students: Student[] = [
+  {
+    id: 0,
+    firstName: "Oski",
+    lastName: "Bear",
+    classes: [classes[0], classes[1]],
+    gpa: 2.1
+  },
+  {
+    id: 1,
+    firstName: "Carol",
+    lastName: "Christ",
+    classes: [classes[1]],
+    gpa: 5.0
+  }
+];
+
+const resolverMap: IResolvers = {
+  Query: {
+    helloWorld(_: void, args: void): string {
+      return `👋 Hello world! 👋`;
+    },
+    allStudents(_: void, args: void): Student[] {
+      return students;
+    },
+    classesById(_: void, args: { id: number }): Class[] | null {
+      const student = students.find(elem => {
+        return elem.id == args.id;
+      });
+      if (student == null) {
+        return null;
+      }
+
+      const studentClasses: Class[] = student.classes!;
+      return classes.filter(elem => {
+        if (studentClasses.includes(elem)) {
+          return elem;
+        }
+      });
+    }
+  },
+  Mutation: {
+    addClass(_: void, args: { id: number; className: string }): Student | null {
+      const student = students.find(elem => {
+        return elem.id == args.id;
+      });
+      if (student == null) {
+        return null;
+      }
+
+      const classFull = classes.find(elem => {
+        return elem.name == args.className;
+      });
+      if (classFull == null) {
+        return null;
+      }
+
+      let studentClasses: Class[] = student.classes!;
+      studentClasses.push(classFull);
+      student.classes = studentClasses;
+      return student;
+    },
+    removeClass(
+      _: void,
+      args: { id: number; className: string }
+    ): Student | null {
+      const student = students.find(elem => {
+        return elem.id == args.id;
+      });
+      if (student == null) {
+        return null;
+      }
+
+      const classFull = classes.find(elem => {
+        return elem.name == args.className;
+      });
+      if (classFull == null) {
+        return null;
+      }
+
+      let studentClasses: Class[] = student.classes!;
```

I think we should check if student.classes exists. If we use `student.classes!`, if there's an empty list it might fail silently